eZ/Publish/Core/MVC/Symfony/FieldType/RichText/Renderer.php

```diff
@@ -69,34 +74,72 @@ class Renderer implements RendererInterface
      */
     private $customTagsConfiguration;
 
+    /**
+     * @var array
+     */
+    private $customStylesConfiguration;
+
     /**
      * @param \eZ\Publish\API\Repository\Repository $repository
      * @param \Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface $authorizationChecker
      * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
      * @param \Symfony\Component\Templating\EngineInterface $templateEngine
      * @param string $tagConfigurationNamespace
+     * @param string $styleConfigurationNamespace
      * @param string $embedConfigurationNamespace
      * @param null|\Psr\Log\LoggerInterface $logger
      * @param array $customTagsConfiguration
+     * @param array $customStylesConfiguration
      */
     public function __construct(
         Repository $repository,
         AuthorizationCheckerInterface $authorizationChecker,
         ConfigResolverInterface $configResolver,
         EngineInterface $templateEngine,
         $tagConfigurationNamespace,
+        $styleConfigurationNamespace,
         $embedConfigurationNamespace,
         LoggerInterface $logger = null,
-        array $customTagsConfiguration = []
+        array $customTagsConfiguration = [],
+        array $customStylesConfiguration = []
     ) {
         $this->repository = $repository;
         $this->authorizationChecker = $authorizationChecker;
         $this->configResolver = $configResolver;
         $this->templateEngine = $templateEngine;
         $this->tagConfigurationNamespace = $tagConfigurationNamespace;
+        $this->styleConfigurationNamespace = $styleConfigurationNamespace;
         $this->embedConfigurationNamespace = $embedConfigurationNamespace;
         $this->logger = $logger;
         $this->customTagsConfiguration = $customTagsConfiguration;
+        $this->customStylesConfiguration = $customStylesConfiguration;
+    }
+
+    public function renderStyle($name, array $parameters, $isInline)
+    {
+        $templateName = $this->getStyleTemplateName($name, $isInline);
+
+        if ($templateName === null) {
+            if (isset($this->logger)) {
+                $this->logger->error(
+                    "Could not render template style '{$name}': no template configured"
+                );
+            }
+
+            return null;
+        }
+
+        if (!$this->templateEngine->exists($templateName)) {
+            if (isset($this->logger)) {
```

Comment:
nitpick, how about a dedicate method for this logging part in order to avoid some dupe code?

Reply:
Agree.
I think i would be best also to move each "render" to dedicated class as this one keep growing.